internal/services/compute/managed_disk_resource.go

```diff
@@ -263,6 +263,11 @@ func resourceManagedDisk() *pluginsdk.Resource {
 				Optional: true,
 			},
 
+			"no_downtime_resize_enabled": {
+				Type:     pluginsdk.TypeBool,
+				Optional: true,
+			},
```

Comment:
since this is a behavioural feature, this'd need to be added to the `features` block rather than being enabled against the resource - although I think we actually want to enable this functionality by default where possible once this goes GA - as such I'm not sure this makes sense as a behavioural feature here, where we can instead wait for this to hit GA first and then enable this by default?

We'll also need to update the `shutDownOnResize` method below to check the disk type too, since this is only supported for Data Disks and not OS Disks

Reply:
Thanks for the review @tombuildsstuff! Implementing your feedback...

This feature is something my team could use immediately (and likely a lot of other users). If possible starting with it being opt-in while in preview would be super helpful to us, especially if GA will take a bit more. After GA we definitely should flip the default of the feature flag as it will make a lot of operations much easier.